--- apache-atlas-2.1.0_ec2/bin/atlas_config.py	2021-06-09 12:24:27.000000000 -0300
+++ atlas_config.py	2021-06-10 10:37:14.000000000 -0300
@@ -509,15 +509,18 @@
 
 def wait_for_startup(confdir, wait):
     count = 0
+    started = False
     host = get_atlas_url_host(confdir)
     port = get_atlas_url_port(confdir)
-    while True:
+    pid_file = pidFile(atlasDir())
+
+    while not started:
         try:
             s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             s.settimeout(1)
             s.connect((host, int(port)))
             s.close()
-            break
+            started = True
         except Exception as e:
             # Wait for 1 sec before next ping
             sys.stdout.write('.')
@@ -526,8 +529,15 @@
 
         if count > wait:
             s.close()
+            sys.stdout.write('\nAtlas Web-UI startup timed out! But, wait for it...')
+            sys.stdout.flush()
             break
 
+        if not os.path.exists(pid_file):
+            sys.stdout.write('\nApache Atlas startup failed!\nCheck logs: /opt/apache-atlas-2.1.0/logs/application.log')
+            sys.stdout.flush()
+            exit()
+
         count = count + 1
 
     sys.stdout.write('\n')
